elastic/kibana
# [Index Templates] Fix edit form when there are missing component templates

Fixes https://github.com/elastic/kibana/issues/186600

## Summary

This PR fixes the following issues in the Index template edit form:

For an index template that is composed of a nonexistent component template and has a `ignore_missing_component_templates` property:
1. when we edit the index template and pass through the "Component template" steps, the nonexistent component template is removed.
2. when we edit the index template and go directly to the "Review" step, the `ignore_missing_templates` field is removed from the request. Therefore, when we try to save the template, it fails with an error as it contains a missing component template and doesn't have a `ignore_missing_templates` property.

**How to test:**
1. Start Es and Kibana
2. Create a test index template composed of a nonexistent component template:
```
PUT _index_template/test
{
  "index_patterns": [
    "test-*"  
  ], 
  "composed_of": [
    "mytesttemplate"
  ],
  "ignore_missing_component_templates": [
    "mytesttemplate"
  ]
}
```
3. Go to Index Management -> Index Templates and start editing the created template.
4. Go directly to the Review step and verify that the request includes both the `composed_of` and the `ignore_missing_component_templates` fields as they were initially. Verify that saving the form doesn't result in an error.
5. Start editing the template again and this time go to the Component Template step. Verify that the non-existent step is displayed in the list. Go to the "Review" step and verify that the request contains the correct `composed_of` and `ignore_missing_component_templates` fields.
6. You can also try the above steps with some of the auto-created index templates that are composed of non-existent component templates (e.g. `logs`, `logs-apm.app@template`, `logs-apm.error@template` - they all are composed of `*@custom` component templates which don't exist).
